pkg/controlplane/instance.go

```diff
@@ -705,10 +666,16 @@ func (m *Instance) InstallAPIs(apiResourceConfigSource serverstorage.APIResource
 			m.GenericAPIServer.AddPostStartHookOrDie(name, hook)
 		}
 
-		apiGroupsInfo = append(apiGroupsInfo, &apiGroupInfo)
+		if len(groupName) == 0 {
+			if err := m.GenericAPIServer.InstallLegacyAPIGroup(genericapiserver.DefaultLegacyAPIPrefix, &apiGroupInfo); err != nil {
+				return fmt.Errorf("error in registering legacy API: %w", err)
+			}
+		} else {
+			nonLegacy = append(nonLegacy, &apiGroupInfo)
+		}
```

Comment:
this is because v1 does not have group, right? 
can we add a comment explaining it if that is the case? is not obvious for people not familiar with this convention

Reply:
Yes, generic apiserver has two API setup calls, for the two path hierarchies. Will add a comment.

Reply:
done